addons/quicktime/fnc_runQTE.sqf

```diff
@@ -0,0 +1,94 @@
+#include "script_component.hpp"
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_runQTE
+
+Description:
+    Runs a Quick-Time Event.
+
+Parameters:
+    _object - <OBJECT>
+    _args - Extra arguments passed to the _on... functions<ARRAY>
+    _onDisplay - Code callback on displayable event passed [_args, _qteSequence, _qteHistory]. <CODE, STRING>
+    _onFinish - Code callback on Quick-Time Event completed passed [_args, _elapsedTime]. <CODE, STRING>
+    _onFinish - Code callback on Quick-Time Event timeout/outranged passed [_args, _elapsedTime]. <CODE, STRING>
+    _qteSequence - Quick-Time sequence made up of ["↑", "↓", "→", "←"] <ARRAY>
+    _maxDistance - max interaction distance from attached object <NUMBER> (default: 10) 
+    _timeout - ingame timeout <NUMBER> (default: 30)
+
+Example:
+    [car,
+    [], 
+    { 
+        hint format [
+            "%1 \n %2",
+            [_this select 1] call CBA_fnc_getFormattedQTESequence,
+            [_this select 2] call CBA_fnc_getFormattedQTESequence
+        ]
+    }, 
+    { 
+        hint "Finished!"; 
+    },
+    { 
+        hint "Failure!"; 
+    },
+    ["↑", "↓", "→", "←"]] call CBA_fnc_runQTE
+
+Returns:
+    True if the QTE was started, false if it was already running <BOOELAN>
+
+Author:
+    john681611
+---------------------------------------------------------------------------- */
+
+
+params ["_object", "_args", "_onDisplay", "_onFinish", "_onFail", "_qteSequence", ["_maxDistance", 10], ["_timeout", 30]];
+if (GVAR(QTERunning)) exitWith {
+    false
+};
+
+GVAR(QTEHistory) = [];
+GVAR(QTERunning) = true;
+private _startTime = CBA_missionTime;
+private _qteArgsArray = [
+    ["object", _object],
+    ["args", _args],
+    ["onDisplay", _onDisplay],
+    ["onFinish", _onFinish],
+    ["onFail", _onFail],
+    ["maxDistance", _maxDistance],
+    ["qteSeqence", _qteSequence],
+    ["startTime", _startTime],
+    ["timeout", _timeout]
+];
+GVAR(QTEArgs) = createHashMapFromArray _qteArgsArray;
+
+// Setup 
+[{
+    private _timeout = GVAR(QTEArgs) get "timeout";
+    private _object = GVAR(QTEArgs) get "object";
+    private _maxDistance = GVAR(QTEArgs) get "maxDistance";
+    private _elapsedTime = CBA_missionTime - (GVAR(QTEArgs) get "startTime");
+    
+    !GVAR(QTERunning) || player distance _object > _maxDistance || _elapsedTime > _timeout;
```

Comment:
I think having `player distance _object > _maxDistance` as a condition is a very limiting factor. In some cases one might not want or need to have such a condition, but would rather use others.
The condition code should be passed by the user, it would be `{false}` by default (meaning the timeout would be the default method of ending the QTE if it isn't solved in time). `object` and `maxDistance` would no longer be needed.

You'd evaluate it by calling `_args call _condition`, with `_args` being the same args as the event/function one.
The function header would need to be adjusted accordingly.

Reply:
Why not go a step further and remove the timeout and just pass  `[_args, _elapsedTime]`. This would allow for unlimited time variants too. Its also standard for all functions then.